packages/driver/src/cy/assertions.coffee

```diff
@@ -56,6 +58,27 @@ parseValueActualAndExpected = (value, actual, expected) ->
 
   obj
 
+prepareObjsForDiff = (err) ->
+  if _.isString(err.actual) || _.isString(err.expected)
+    return err
+  ret = {}
+  ret.actual = mochaUtils.stringify(err.actual)
+  ret.expected = mochaUtils.stringify(err.expected)
+  return ret
+
+objToString = Object.prototype.toString
+
+_sameType = (a, b) ->
+  return objToString.call(a) is objToString.call(b)
```

Comment:
I'm in favor of changing this to a `typeof` check. Any thoughts? @bahmutov 

that way, a diff will show up as long as the js type matches, not the object's class name

AKA: `MouseEvent {clientX: 39, clientY:50}` should diff with `Object {clientX:40, clientY:50}`

using typeof these would be the same type, Object.prototype.toString they will not be